src/System.Windows.Forms/src/System/Windows/Forms/ListView.ListViewAccessibleObject.cs

```diff
@@ -0,0 +1,260 @@
+                    return true;
+                }
+
+                return base.IsPatternSupported(patternId);
+            }
+
+            internal override IRawElementProviderFragment FragmentNavigate(UiaCore.NavigateDirection direction)
+            {
+                int childCount = OwningListView.Items.Count;
+
+                if (childCount == 0)
+                {
+                    return null;
+                }
+
+                switch (direction)
+                {
+                    case NavigateDirection.FirstChild:
+                        return GetChild(0);
+                    case NavigateDirection.LastChild:
+                        return GetChild(childCount - 1);
+                    default:
+                        return base.FragmentNavigate(direction);
+                }
+            }
+
+            internal override bool CanSelectMultiple => true;
+
+            internal override object GetPropertyValue(UIA propertyID)
+            {
+                switch (propertyID)
+                {
+                    case UIA.NamePropertyId:
+                        return Name;
+                    case UIA.AutomationIdPropertyId:
+                        return OwningListView.IsHandleCreated ? OwningListView.Name : string.Empty;
+                    case UIA.RuntimeIdPropertyId:
+                        return RuntimeId;
+                    case UIA.ControlTypePropertyId:
+                        return UIA.ListControlTypeId;
+                    case UIA.IsMultipleViewPatternAvailablePropertyId:
+                        return IsPatternSupported(UIA.MultipleViewPatternId);
+                    case UIA.ItemStatusPropertyId:
+                        return GetItemStatus();
+                }
+
+                return base.GetPropertyValue(propertyID);
+            }
+
+            internal override int[] RuntimeId
+            {
+                get
+                {
+                    var runtimeId = new int[2];
+                    runtimeId[0] = 0x2a;
+                    runtimeId[1] = (int)(long)OwningListView.Handle;
+                    return runtimeId;
+                }
+            }
+
+            private string GetItemStatus()
+            {
+                switch (owner.Sorting)
+                {
+                    case SortOrder.None:
+                        return SR.NotSortedAccessibleStatus;
+                    case SortOrder.Ascending:
+                        return SR.SortedAscendingAccessibleStatus;
+                    case SortOrder.Descending:
+                        return SR.SortedDescendingAccessibleStatus;
+                }
+
+                return null;
+            }
+
+            public override AccessibleObject GetChild(int index)
+            {
+                if (index < 0 || index >= OwningListView.Items.Count)
+                {
+                    return null;
+                }
+
+                ListViewItem item = OwningListView.Items[index];
```

Comment:
This looks like only ListViewItems can be children of the ListView. However Groups are displayed in the ListView Hierarchy as well. Should we add support for them?

Reply:
Yes, thanks, we should. I'll add it.